tools/Azure.Mcp.Tools.Foundry/src/Services/FoundryService.cs

```diff
@@ -323,4 +349,540 @@ public async Task<KnowledgeIndexSchema> GetKnowledgeIndexSchema(string endpoint,
+            {
+                { "success", true },
+                { "thread_id", threadId },
+                { "run_id", runId },
+                { "text_response", textResponse.ToString().Trim() },
+                { "text_query", query },
+                { "agent_id", agentId },
+                { "response", convertedResponse },
+                { "query", convertedRequestMessages },
+                { "tool_definitions", JsonSerializer.Serialize(toolDefinitions, (JsonTypeInfo<List<ToolDefinitionAIFunction>>)AIJsonUtilities.DefaultOptions.GetTypeInfo(typeof(List<ToolDefinitionAIFunction>))) },
+                { "citations", citations }
+            };
+        }
+        catch (Exception ex)
+        {
+            throw new Exception($"Failed to connect to agent: {ex.Message}", ex);
+        }
+    }
+
+    public async Task<Dictionary<string, object>> QueryAndEvaluateAgent(string agentId, string query, string endpoint, string azureOpenAIEndpoint, string azureOpenAIDeployment, string? tenant = null, List<string>? evaluatorNames = null, RetryPolicyOptions? retryPolicy = null)
+    {
+        try
+        {
+            ValidateRequiredParameters(agentId, query, endpoint);
+
+            var connectAgentResult = await ConnectAgent(agentId, query, endpoint, tenant, retryPolicy);
+
+            var credential = await GetCredential(tenant);
+
+            List<IEvaluator> evaluators = [];
+            List<EvaluationContext> evaluationContexts = [];
+            var toolDefinitions = connectAgentResult["tool_definitions"] as string;
+            var loadedToolDefinitions = ConvertToolDefinitionsFromString(toolDefinitions);
+
+            if (evaluatorNames == null || evaluatorNames.Count == 0)
+            {
+                evaluatorNames = AgentEvaluatorDictionary.Keys.ToList();
+            }
+
+            foreach (var name in evaluatorNames)
+            {
+                var evaluatorName = name.ToLowerInvariant();
+                if (AgentEvaluatorDictionary.TryGetValue(evaluatorName, out var createEvaluator))
+                {
+                    var evaluator = createEvaluator();
+                    evaluators.Add(evaluator);
+                    evaluationContexts.Add(AgentEvaluatorContextDictionary[evaluatorName](loadedToolDefinitions ?? []));
+                }
+            }
+            var compositeEvaluator = new CompositeEvaluator(evaluators);
+
+            var azureOpenAIChatClient = GetAzureOpenAIChatClient(azureOpenAIEndpoint, azureOpenAIDeployment, credential);
+
+            var evaluationResult = await compositeEvaluator.EvaluateAsync(
+                connectAgentResult["query"] as List<ChatMessage> ?? [],
+                new ChatResponse(connectAgentResult["response"] as List<ChatMessage> ?? []),
+                new ChatConfiguration(azureOpenAIChatClient),
+                evaluationContexts);
+
+            return new Dictionary<string, object>
+            {
+                { "success", true },
```

Comment:
You may want to refactor this to use serialization or reuse the binding logic across methods.